MSO provider plugin crashes when trying to apply simple filter entry

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

Terraform v1.2.9
on darwin_arm64
+ provider registry.terraform.io/ciscodevnet/mso v0.7.0

### MSO version

* V 4.0(2i)

### APIC version and APIC Platform for Site Level Resources

* V 6.0(1g)

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* mso_schema_template_filter_entry

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```
resource "mso_schema_template_filter_entry" "filter1" {
  schema_id = mso_schema.schema-main.id
  template_name = "Common-Items"
  name = "ICMP"
  display_name = "ICMP"
  entry_name = "ping"
  entry_display_name = "ping"
  ether_type = "ip"
  ip_protocol = "icmp"
  destination_from = "unspecified"
  destination_to = "unspecified"
  source_from = "unspecified"
  source_to = "unspecified"
}

### Debug Output

Error: Plugin did not respond
│ 
│   with mso_schema_template_filter_entry.filter1,
│   on NDO-Basics.tf line 73, in resource "mso_schema_template_filter_entry" "filter1":
│   73: resource "mso_schema_template_filter_entry" "filter1" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
│ more details.

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

Stack trace from the terraform-provider-mso_v0.7.0 plugin:

panic: interface conversion: interface {} is nil, not []interface {}

goroutine 85 [running]:
github.com/terraform-providers/terraform-provider-mso/mso.resourceMSOSchemaTemplateFilterEntryRead(0x140004e51f0, {0x102f17ca0, 0x14000734d00})
        github.com/terraform-providers/terraform-provider-mso/mso/resource_mso_schema_template_filter_entry.go:458 +0x1698
github.com/terraform-providers/terraform-provider-mso/mso.resourceMSOSchemaTemplateFilterEntryCreate(0x140004e51f0, {0x102f17ca0, 0x14000734d00})
        github.com/terraform-providers/terraform-provider-mso/mso/resource_mso_schema_template_filter_entry.go:368 +0x140c
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0x1400068de00, 0x1400090b040, 0x14000919120, {0x102f17ca0, 0x14000734d00})
        github.com/hashicorp/terraform-plugin-sdk@v1.17.1/helper/schema/resource.go:320 +0x528
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0x14000312980, 0x140004c5a40, 0x1400090b040, 0x14000919120)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.1/helper/schema/provider.go:294 +0x88
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0x14000306988, {0x102f76f50, 0x1400083b4a0}, 0x140004e48c0)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.1/internal/helper/plugin/grpc_provider.go:895 +0x8cc
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x102f24540, 0x14000306988}, {0x102f76f50, 0x1400083b4a0}, 0x14000198a20, 0x0)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.1/internal/tfplugin5/tfplugin5.pb.go:3305 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140003c5880, {0x102f85bf8, 0x14000486600}, 0x140006f2600, 0x140002dea20, 0x103598a00, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0xc38
google.golang.org/grpc.(*Server).handleStream(0x140003c5880, {0x102f85bf8, 0x14000486600}, 0x140006f2600, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xa34
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x14000034190, 0x140003c5880, {0x102f85bf8, 0x14000486600}, 0x140006f2600)
        google.golang.org/grpc@v1.32.0/server.go:859 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1f0

Error: The terraform-provider-mso_v0.7.0 plugin crashed!

### Expected Behavior

<!--- What should have happened? --->
Complete creation of filter in NDO

### Actual Behavior
FIlter only partially deployed in NDO and subsequent tf plugin crash
<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/mso/datasource_mso_schema_template_filter_entry.go
+++ b/mso/datasource_mso_schema_template_filter_entry.go
@@ -167,7 +167,7 @@ func dataSourceMSOSchemaTemplateFilterEntryRead(d *schema.ResourceData, m interf
 							d.SetId(apiFilterEntry)
 							d.Set("entry_name", apiFilterEntry)
 							d.Set("entry_display_name", models.StripQuotes(entriesCont.S("displayName").String()))
-							if entriesCont.Exists("descirption") {
+							if entriesCont.Exists("description") {
 								d.Set("entry_description", models.StripQuotes(entriesCont.S("description").String()))
 							}
 							if entriesCont.Exists("etherType") {

--- a/mso/resource_mso_schema_template_filter_entry.go
+++ b/mso/resource_mso_schema_template_filter_entry.go
@@ -175,7 +175,7 @@ func resourceMSOSchemaTemplateFilterEntryImport(d *schema.ResourceData, m interf
 							d.SetId(apiFilterEntry)
 							d.Set("entry_name", apiFilterEntry)
 							d.Set("entry_display_name", models.StripQuotes(entriesCont.S("displayName").String()))
-							if entriesCont.Exists("descirption") {
+							if entriesCont.Exists("description") {
 								d.Set("entry_description", models.StripQuotes(entriesCont.S("description").String()))
 							}
 							if entriesCont.Exists("etherType") {
@@ -289,6 +289,8 @@ func resourceMSOSchemaTemplateFilterEntryCreate(d *schema.ResourceData, m interf
 	if tempVar, ok := d.GetOk("tcp_session_rules"); ok {
 		tcpSessionRules = tempVar.([]interface{})
 		entryMap["tcpSessionRules"] = tcpSessionRules
+	} else {
+		entryMap["tcpSessionRules"] = append(make([]interface{}, 0), "unspecified")
 	}
 	entries = append(entries, entryMap)
 	foundEntry := false
@@ -424,7 +426,7 @@ func resourceMSOSchemaTemplateFilterEntryRead(d *schema.ResourceData, m interfac
 							d.SetId(apiFilterEntry)
 							d.Set("entry_name", apiFilterEntry)
 							d.Set("entry_display_name", models.StripQuotes(entriesCont.S("displayName").String()))
-							if entriesCont.Exists("descirption") {
+							if entriesCont.Exists("description") {
 								d.Set("entry_description", models.StripQuotes(entriesCont.S("description").String()))
 							}
 							if entriesCont.Exists("etherType") {
@@ -456,8 +458,6 @@ func resourceMSOSchemaTemplateFilterEntryRead(d *schema.ResourceData, m interfac
 							}
 							if entriesCont.Exists("tcpSessionRules") {
 								d.Set("tcp_session_rules", entriesCont.S("tcpSessionRules").Data().([]interface{}))
-							} else {
-								d.Set("tcp_session_rules", make([]interface{}, 0))
 							}
 							found = true
 							break
@@ -492,47 +492,38 @@ func resourceMSOSchemaTemplateFilterEntryUpdate(d *schema.ResourceData, m interf
 
 	if tempVar, ok := d.GetOk("entry_description"); ok {
 		entryDescription = tempVar.(string)
-
 	}
 	if tempVar, ok := d.GetOk("ether_type"); ok {
 		etherType = tempVar.(string)
-
 	}
 	if tempVar, ok := d.GetOk("arp_flag"); ok {
 		arpFlag = tempVar.(string)
-
 	}
 	if tempVar, ok := d.GetOk("ip_protocol"); ok {
 		ipProtocol = tempVar.(string)
-
 	}
 	if tempVar, ok := d.GetOk("source_from"); ok {
 		sourceFrom = tempVar.(string)
-
 	}
 	if tempVar, ok := d.GetOk("source_to"); ok {
 		sourceTo = tempVar.(string)
-
 	}
 	if tempVar, ok := d.GetOk("destination_from"); ok {
 		destinationFrom = tempVar.(string)
 	}
 	if tempVar, ok := d.GetOk("destination_to"); ok {
-
 		destinationTo = tempVar.(string)
 	}
 	if tempVar, ok := d.GetOk("match_only_fragments"); ok {
 		matchOnlyFragments = tempVar.(bool)
-
 	}
 	if tempVar, ok := d.GetOk("stateful"); ok {
 		stateful = tempVar.(bool)
-
 	}
 	if tempVar, ok := d.GetOk("tcp_session_rules"); ok {
-
 		tcpSessionRules = tempVar.([]interface{})
-
+	} else {
+		tcpSessionRules = append(make([]interface{}, 0), "unspecified")
 	}
 
 	pathf := fmt.Sprintf("/templates/%s/filters/%s/entries/%s", stateTemplate, filterName, entryName)
@@ -605,7 +596,6 @@ func resourceMSOSchemaTemplateFilterEntryDelete(d *schema.ResourceData, m interf
 		entryMap["stateful"] = stateful
 	}
 	if tempVar, ok := d.GetOk("tcp_session_rules"); ok {
-
 		tcpSessionRules = tempVar.([]interface{})
 		entryMap["tcpSessionRules"] = tcpSessionRules
 	}

